tests/failpoints/cases/test_split_region.rs

```diff
@@ -552,6 +552,90 @@ fn test_split_not_to_split_existing_tombstone_region() {
     must_get_equal(&cluster.get_engine(2), b"k1", b"v1");
 }
 
+#[test]
+fn test_snap_handling_after_peer_is_replaced_by_split_and_removed() {
```

Comment:
Could you add another test for handling raft heartbeat, append and request vote messages after old peer is replaced?

Reply:
Added such tests. I tried my best to reuse existing code.

Reply:
One thing to note: only the snapshot message test case caused a panic, other tests passed even without the fix.